### PR TITLE
Fix concatenating of surrogates.

### PR DESCRIPTION
**Pull request by sand1k**
This patch fixes the issue #346.

### CODE REVIEW

File: jerry-core/ecma/base/ecma-helpers-string.cpp [hunk 1]
```diff
@@ -1039,13 +1070,37 @@ ecma_string_to_utf8_string (const ecma_string_t *string_desc_p, /**< ecma-string
...
+        ecma_char_t low_surrogate = lit_utf8_string_code_unit_at (dest_p + 1, LIT_UTF8_MAX_BYTES_IN_CODE_UNIT, 0);
+        JERRY_ASSERT (lit_is_code_unit_low_surrogate (low_surrogate));
+
+        lit_code_point_t surrogate_code_point = lit_convert_surrogate_pair_to_code_point (high_surrogate,
+                                                                                          low_surrogate);
+        lit_code_point_to_utf8 (surrogate_code_point, dest_p);
+      }
 
-      JERRY_ASSERT (required_buffer_size == bytes_copied1 + bytes_copied2);
+      JERRY_ASSERT (required_buffer_size == (bytes_copied1 + bytes_copied2
```
Comment (egavrin) [inline jerry-core/ecma/base/ecma-helpers-string.cpp]: Style is a bit different:
```
+      string_desc_p->hash = lit_utf8_string_calc_hash_last_bytes (bytes_buf + bytes_buf_size -
+                                                                  LIT_STRING_HASH_LAST_BYTES_COUNT,
+                                                                  LIT_STRING_HASH_LAST_BYTES_COUNT);
```

Comment (egavrin) [inline jerry-core/ecma/base/ecma-helpers-string.cpp]: @sand1k we need style checker.

**Comment by egavrin** [comment]
Good to me

File: jerry-core/lit/lit-strings.h [hunk 2]
```diff
@@ -73,6 +73,8 @@
 #define LIT_UTF8_4_BYTE_CODE_POINT_MIN (0x10000)
 #define LIT_UTF8_4_BYTE_CODE_POINT_MAX (LIT_UNICODE_CODE_POINT_MAX)
 
+#define LIT_UTF8_CESU8_SURROGATE_SIZE_DIF (2 * LIT_UTF8_MAX_BYTES_IN_CODE_UNIT - LIT_UTF8_MAX_BYTES_IN_CODE_POINT)
```
Comment (ruben-ayrapetyan) [inline jerry-core/lit/lit-strings.h]: @sand1k, could you, please add comment with description for the value?

**Comment by zherczeg** [comment]
+1 from me

**Pull request merged**